src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -375,6 +373,12 @@ protected function getAttributeFromArray($key)
         if (isset($this->attributes[$key])) {
             return $this->attributes[$key];
         }
+
+        if (Str::contains($key, '->')) {
+            [$key, $path] = explode('->', $key, 2);
+
+            return Arr::get($this->getArrayAttributeByKey($key), str_replace('->', '.', $key));
```

Comment:
Your code doesn't work:

```suggestion
            return Arr::get($this->getArrayAttributeByKey($key), str_replace('->', '.', $path));
```

Reply:
Oh, thanks for the catch, copy/paste typo, tests will catch all this things at the end.